Tests/Algorithm/Framework/Portfolio/MaximumSharpeRatioPortfolioOptimizerTests.cs

```diff
@@ -0,0 +1,200 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by aaplicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using NUnit.Framework;
+using QuantConnect.Algorithm.Framework.Portfolio;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Tests.Algorithm.Framework.Portfolio
+{
+    [TestFixture]
+    public class MaximumSharpeRatioPortfolioOptimizerTests
+    {
+        private Dictionary<int, double[,]> _historicalReturns = new();
+        private Dictionary<int, double[]> _expectedReturns = new();
+        private Dictionary<int, double[,]> _covariances = new();
+        private Dictionary<int, double[]> _expectedResults = new();
+
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            double[,] historicalReturns1 = new double[,] { { 0.02, -0.02, 0.28 }, { -0.50, -0.29, -0.13 }, { 0.81, 0.29, 0.31 }, { -0.03, -0.00, 0.01 } };
```

Comment:
In Lean it is a common practice to use prioritize the use of `var` vs. explicitly typing the variable